### PR TITLE
BSM: Add component and rule template to supported resource types for sensuctl

### LINKED ISSUE #2951: Update sensuctl resource types lists for BSM
### Description
Sensuctl docs include lists of resource types that work with each sensuctl command. These may need to be updated with `RuleTemplate` and `Component`.

### Category

This is an issue with:

- [ ] errata (error in doc content)
- [ ] new doc (requires a completely new doc)
- [x] update (add missing or refresh existing content)
- [ ] improvement (improve docs content or format)

### Affected pages
https://docs.sensu.io/sensu-go/latest/sensuctl/create-manage-resources/#supported-resource-types (prune)
https://docs.sensu.io/sensu-go/latest/sensuctl/back-up-recover/#supported-resource-types (dump)

### Expected behavior
Resource types are accurate for sensuctl dump and prune commands

### Current behavior
No sensuctl resource type lists include `RuleTemplate` or `ServiceComponent`.

### Corrections needed
Update sensuctl resource type lists to include `RuleTemplate` or `ServiceComponent` if appropriate.

### Additional context
[BSM proposal](https://docs.google.com/document/d/1eMwAt2oHLh-lrWev6YicyiuzKvFF-dx7FWXV5DWem9Y/edit#)

### PR DESCRIPTION
**Pull request by hillaryfraley**
## Description
Adds the BSM resources, component and rule template, to the lists of supported resource types for sensuctl prune.

## Motivation and Context
Closes https://github.com/sensu/sensu-docs/issues/2951

### CODE REVIEW

File: content/sensu-go/6.3/sensuctl/create-manage-resources.md [hunk 1]
```diff
@@ -244,10 +245,10 @@ Requests to update agent-managed entities via sensuctl will fail and return an e
...
 --------------------|---|---|---|
 `asset` | `check` | `cluster` | `cluster-role`
-`cluster-role-binding` | `entity` | `event` | `filter`
-`handler` | `hook` | `mutator` | `namespace`
-`role` | `role-binding` | `silenced` | `user`
-[`auth`][26] | | |
+`cluster-role-binding` | `component` | `entity` | `event`
+`filter` | `handler` | `hook` | `mutator`
+`namespace` | `role` | `role-binding` | `rule-template`
+`silenced` | `user` | [`auth`][26] | |
```
Comment (hillaryfraley) [inline content/sensu-go/6.3/sensuctl/create-manage-resources.md]: I'm not sure if lower-case "component" and lower-case and hyphenated "rule-template" is correct here for sensuctl edit. Just guessing -- please confirm

File: content/sensu-go/6.3/sensuctl/create-manage-resources.md [hunk 2]
```diff
@@ -145,15 +145,16 @@ The following table describes the command-specific flags.
...
+`ClusterRoleBinding`  | `cluster_role_binding` | [`Component`][38] | `Entity`
+[`Env`][24] | `entity` | [`EtcdReplicators`][29] | `Event`
+`event`| `EventFilter` | `event_filter` | [`GlobalConfig`][11]
+`Handler`  | `handler` | `Hook` | `hook`
+`HookConfig` | `hook_config` | `Mutator` | `mutator`
+`Namespace` | `namespace` | `Role` | `role`
+`RoleBinding` | `role_binding` | [`RuleTemplate`][39] | [`Secret`][28]
+`Silenced` | `silenced` | [`User`][8] | `user`
+[`VaultProvider`][24] | [`ldap`][26] | [`ad`][25] | [`oidc`][37]
+[`TessenConfig`][27] | [`PostgresConfig`][32]
```
Comment (hillaryfraley) [inline content/sensu-go/6.3/sensuctl/create-manage-resources.md:157]: Linked pages are not implemented yet but will be the Components reference and Rule templates reference, respectively.

**Comment by hillaryfraley** [comment]
Still unclear whether the correct component type is `Component` or `ServiceComponent`

**Comment by hillaryfraley** [comment]
Do rule templates and components belong in this list too? https://docs.sensu.io/sensu-go/latest/sensuctl/create-manage-resources/#manage-resources

File: content/sensu-go/6.3/sensuctl/create-manage-resources.md (hunk 1 above)
Comment (ccressent) [reply content/sensu-go/6.3/sensuctl/create-manage-resources.md]: I don't think there are "shortcuts" defined for them, so one needs to use the fully qualified name as listed by `sensuctl describe-type all`: `bsm/v1.RuleTemplate` and `bsm/v1.ServiceComponent`.

Comment (hillaryfraley) [reply content/sensu-go/6.3/sensuctl/create-manage-resources.md]: Where can I look in the code to find the resource names to use with difference sensuctl commands? None of the rule template or service component permutations I've tried seem to work. I think more generally these sensuctl lists are in some disarray and I'd like to fix them. Or is there a sensuctl command that lists the resource names to use with each command type, kind of like the describe-type list but for create, edit, etc?

**Comment by hillaryfraley** [comment]
Sounds like BSM resources are not supported for sensuctl commands outside describe-type and dump: https://sensu.slack.com/archives/C60EEQFH8/p1616508141272600

File: content/sensu-go/6.3/sensuctl/create-manage-resources.md (hunk 1 above)
Comment (hillaryfraley) [reply content/sensu-go/6.3/sensuctl/create-manage-resources.md]: Disregard -- see https://sensu.slack.com/archives/C60EEQFH8/p1616508141272600

File: content/sensu-go/6.3/sensuctl/create-manage-resources.md [hunk 3]
```diff
@@ -673,3 +675,5 @@ Sensuctl supports the following formats:
 [35]: ../../operations/control-access/rbac/#roles-and-cluster-roles
 [36]: #sensuctl-create-flags
 [37]: ../../operations/control-access/oidc-auth/
+[38]: ../../observability-pipeline/observe-schedule/service-components/
+[39]: ../../observability-pipeline/observe-schedule/rule-templates/
```
Comment (hillaryfraley) [inline content/sensu-go/6.3/sensuctl/create-manage-resources.md]:
```suggestion
```

**Review by ccressent — APPROVED** [review]

**Pull request merged**